src/graphql/node/control.rs

```diff
@@ -74,14 +91,924 @@ impl NodeControlMutation {
             )
         }
     }
+
+    #[graphql(guard = "RoleGuard::new(Role::SystemAdministrator)
+        .or(RoleGuard::new(Role::SecurityAdministrator))")]
+    async fn apply_node(&self, ctx: &Context<'_>, id: ID) -> Result<ApplyResult> {
+        let i = id.as_str().parse::<u32>().map_err(|_| "invalid ID")?;
+        let agents = ctx.data::<BoxedAgentManager>()?;
+
+        let node: Node = {
+            let store = crate::graphql::get_store(ctx).await?;
+            let node_map = store.node_map();
+            node_map
+                .get_by_id(i)?
+                .ok_or_else(|| async_graphql::Error::new(format!("Node with ID {i} not found",)))?
+        };
+
+        if node.name_draft.is_none() && node.settings_draft.is_none() {
```

Comment:
What value will `name_draft` have when users don't want to change `name`? I thought it would have `None`, which is, I think, the way we have been doing.

Reply:
I am sorry I misunderstood. Forget about it.